### PR TITLE
Create Telegraf commands and flags doc

### LINKED ISSUE #3032: Add Telegraf commands and flags to docs
Telegraf commands and flags aren't documented anywhere. I have a [PR](https://github.com/influxdata/telegraf/pull/9635) to add them to the GitHub docs but they should be in the customer facing docs as well. 

GitHub docs:
- https://github.com/influxdata/telegraf/blob/ab15e5797bcb80e4f44df3f8d09bf8e5267a8a1e/docs/COMMANDS_AND_FLAGS.md

Something similar to what we have for `influxd`: 
- https://docs.influxdata.com/influxdb/v2.0/reference/cli/influxd/

### PR DESCRIPTION
**Pull request by noramullen1**
Closes #3032 

@sanderson , any thoughts on where this should live? Current location isn't perfect but I wasn't sure where else.

### CODE REVIEW

**Comment by sjwang90** [comment]
I think we still need to include the `<file>` `<filter>` etc in the left column. the user won't get what they want if they don't include whatever is in <>

**Comment by noramullen1** [comment]
> I think we still need to include the `<file>` `<filter>` etc in the left column. the user won't get what they want if they don't include whatever is in <>

Oops! I didn't mean to exclude them, they got stripped out in the table generator i used. Thanks!

**Review by sjwang90 — APPROVED** [review]

File: content/telegraf/v1.20/administration/commands.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+## Commands
+
++-----------+------------------------------------------------+
+| Command   | Description                                    |
++-----------+------------------------------------------------+
+| `config`  | Print out full sample configuration to stdout. |
++-----------+------------------------------------------------+
+| `version` | Print version to stdout.                       |
++-----------+------------------------------------------------+
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:28]: This isn't a valid markdown table.
```suggestion


| Command   | Description                                    |
| :-------- | :--------------------------------------------- |
| `config`  | Print out full sample configuration to stdout. |
| `version` | Print version to stdout.                       |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
++------------------------+--------------------------------------------------------------------------------------------------------------------------------+
+| `--test-wait`          | Wait up to the specified number of seconds for service inputs to complete in test or once mode.                                |
++------------------------+--------------------------------------------------------------------------------------------------------------------------------+
+| `--usage <plugin>`              | Print usage for a plugin. For example, `telegraf --usage mysql`.                                                               |
++------------------------+--------------------------------------------------------------------------------------------------------------------------------+
+| `--version`            | Display the version and exit.                                                                                                  |
++------------------------+--------------------------------------------------------------------------------------------------------------------------------+
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:80]:
```suggestion
| Flag                             | Description                                                                                                                       |
| :------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------- |
| `--aggregator-filter <filter>`   | Filter aggregators to enable. Separator is `:`.                                                                                   |
| `--config <file>`                | Configuration file to load.                                                                                                       |
| `--config-directory <directory>` | Directory containing additional `*.conf` files                                                                                    |
| `--watch-config`                 | Restart Telegraf on local config changes. Use either fs notifications (`inotify`) or polling (`poll`). Disabled by default        |
| `--plugin-directory <directory>` | Directory containing `*.so` files to search recursively for plugins. Found plugins are loaded, tagged, and identified.            |
| `--debug`                        | Enable debug logging.                                                                                                             |
| `--input-filter <filter>`        | Filter inputs to enable. Separator is `:`.                                                                                        |
| `--input-list`                   | Print available input plugins.                                                                                                    |
| `--output-filter`                | Filter outputs to enable. Separator is `:`.                                                                                       |
| `--output-list`                  | Print available output plugins.                                                                                                   |
| `--pidfile <file>`               | File to write PID to.                                                                                                             |
| `--pprof-addr <address>`         | pprof address to listen on. Disabled by default.                                                                                  |
| `--processor-filter <filter>`    | Filter processors to enable. Separator is `:`.                                                                                    |
| `--quiet`                        | Run in quiet mode.                                                                                                                |
| `--section-filter <filter>`      | Filter config sections to output (`agent`, `global_tags`, `outputs`, `processors`, `aggregators` and `inputs`). Separator is `:`. |
| `--sample-config`                | Print full sample configuration.                                                                                                  |
| `--once`                         | Gather metrics once, write them, and exit.                                                                                        |
| `--test`                         | Gather metrics once and print them.                                                                                               |
| `--test-wait`                    | Number of seconds to wait for service inputs to complete in test or once mode.                                                    |
| `--usage <plugin>`               | Print plugin usage (example: `telegraf --usage mysql`).                                                                           |
| `--version`                      | Print Telegraf version                                                                                                            |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+## Examples
+
+### Generate a telegraf config file
+
+`telegraf config > telegraf.conf`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:86]:
~~~suggestion
```sh
telegraf config > telegraf.conf
```
~~~

File: content/telegraf/v1.20/administration/commands.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+### Generate config with only CPU input and InfluxDB output plugins defined
+
+`telegraf --input-filter cpu --output-filter influxdb config`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:90]:
~~~suggestion
```sh
telegraf --input-filter cpu --output-filter influxdb config
```
~~~

File: content/telegraf/v1.20/administration/commands.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+### Run a single Telegraf collection, outputting metrics to stdout
+
+`telegraf --config telegraf.conf --test`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:94]:
~~~suggestion
```sh
telegraf --config telegraf.conf --test
```
~~~

File: content/telegraf/v1.20/administration/commands.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+### Run telegraf with all plugins defined in config file:**
+
+`telegraf --config telegraf.conf`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:98]:
~~~suggestion
```sh
telegraf --config telegraf.conf
```
~~~

File: content/telegraf/v1.20/administration/commands.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+### Run Telegraf, enabling the CPU and memory inputs and InfluxDB output plugins:**
+
+`telegraf --config telegraf.conf --input-filter cpu:mem --output-filter influxdb`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:102]:
~~~suggestion
```sh
telegraf --config telegraf.conf --input-filter cpu:mem --output-filter influxdb
```
~~~

File: content/telegraf/v1.20/administration/commands.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+### Run Telegraf with pprof
+
+`telegraf --config telegraf.conf --pprof-addr localhost:6060`
```
Comment (sanderson) [inline content/telegraf/v1.20/administration/commands.md:106]:
~~~suggestion
```sh
telegraf --config telegraf.conf --pprof-addr localhost:6060
```
~~~

**Review by sanderson — CHANGES_REQUESTED** [review]
I think the location is fine. I added suggestions for formatting changes.

File: content/telegraf/v1.20/administration/commands.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+| `config`  | Print out full sample configuration to stdout. |
+| `version` | Print version to stdout.                       |
+
+## Flags
+
+| Flag                             | Description                                                                                                                       |
+| :------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------- |
+| `--aggregator-filter <filter>`   | Filter aggregators to enable. Separator is `:`.                                                                                   |
+| `--config <file>`                | Configuration file to load.                                                                                                       |
+| `--config-directory <directory>` | Directory containing additional `*.conf` files                                                                                    |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:35]:
```suggestion
| `--config-directory <directory>` | Directory containing additional `*.conf` files.                                                                                    |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+| `--section-filter <filter>`      | Filter config sections to output (`agent`, `global_tags`, `outputs`, `processors`, `aggregators` and `inputs`). Separator is `:`. |
+| `--sample-config`                | Print full sample configuration.                                                                                                  |
+| `--once`                         | Gather metrics once, write them, and exit.                                                                                        |
+| `--test`                         | Gather metrics once and print them.                                                                                               |
+| `--test-wait`                    | Number of seconds to wait for service inputs to complete in test or once mode.                                                    |
+| `--usage <plugin>`               | Print plugin usage (example: `telegraf --usage mysql`).                                                                           |
+| `--version`                      | Print Telegraf version                                                                                                            |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:53]:
```suggestion
| `--version`                      | Print Telegraf version.                                                                                                            |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+## Examples
+
+### Generate a telegraf config file
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:57]:
```suggestion
### Generate a Telegraf config file
```

File: content/telegraf/v1.20/administration/commands.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+telegraf --input-filter cpu --output-filter influxdb config
+```
+
+### Run a single Telegraf collection, outputting metrics to stdout
+
+```sh
+telegraf --config telegraf.conf --test
+```
+
+### Run telegraf with all plugins defined in config file:**
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:75]:
```suggestion
### Run Telegraf with all plugins defined in config file**
```

File: content/telegraf/v1.20/administration/commands.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+```sh
+telegraf --config telegraf.conf
+```
+
+### Run Telegraf, enabling the CPU and memory inputs and InfluxDB output plugins:**
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:81]:
```suggestion
### Run Telegraf, enabling the CPU and memory input plugins and InfluxDB output plugin**
```

File: content/telegraf/v1.20/administration/commands.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+| `--processor-filter <filter>`    | Filter processors to enable. Separator is `:`.                                                                                    |
+| `--quiet`                        | Run in quiet mode.                                                                                                                |
+| `--section-filter <filter>`      | Filter config sections to output (`agent`, `global_tags`, `outputs`, `processors`, `aggregators` and `inputs`). Separator is `:`. |
+| `--sample-config`                | Print full sample configuration.                                                                                                  |
+| `--once`                         | Gather metrics once, write them, and exit.                                                                                        |
+| `--test`                         | Gather metrics once and print them.                                                                                               |
+| `--test-wait`                    | Number of seconds to wait for service inputs to complete in test or once mode.                                                    |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:51]:
```suggestion
| `--test-wait`                    | Number of seconds to wait for service inputs to complete in `test` or `once` mode.                                                    |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+| `--aggregator-filter <filter>`   | Filter aggregators to enable. Separator is `:`.                                                                                   |
+| `--config <file>`                | Configuration file to load.                                                                                                       |
+| `--config-directory <directory>` | Directory containing additional `*.conf` files                                                                                    |
+| `--watch-config`                 | Restart Telegraf on local config changes. Use either fs notifications (`inotify`) or polling (`poll`). Disabled by default        |
+| `--plugin-directory <directory>` | Directory containing `*.so` files to search recursively for plugins. Found plugins are loaded, tagged, and identified.            |
+| `--debug`                        | Enable debug logging.                                                                                                             |
+| `--input-filter <filter>`        | Filter inputs to enable. Separator is `:`.                                                                                        |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:39]:
```suggestion
| `--input-filter <filter>`        | Filter input plugins to enable. Separator is `:`.                                                                                        |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+| `--input-list`                   | Print available input plugins.                                                                                                    |
+| `--output-filter`                | Filter outputs to enable. Separator is `:`.                                                                                       |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:41]:
```suggestion
| `--output-filter`                | Filter output plugins to enable. Separator is `:`.                                                                                       |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+| `--output-list`                  | Print available output plugins.                                                                                                   |
+| `--pidfile <file>`               | File to write PID to.                                                                                                             |
+| `--pprof-addr <address>`         | pprof address to listen on. Disabled by default.                                                                                  |
+| `--processor-filter <filter>`    | Filter processors to enable. Separator is `:`.                                                                                    |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:45]:
```suggestion
| `--processor-filter <filter>`    | Filter processor plugins to enable. Separator is `:`.                                                                                    |
```

File: content/telegraf/v1.20/administration/commands.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+| `version` | Print version to stdout.                       |
+
+## Flags
+
+| Flag                             | Description                                                                                                                       |
+| :------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------- |
+| `--aggregator-filter <filter>`   | Filter aggregators to enable. Separator is `:`.                                                                                   |
+| `--config <file>`                | Configuration file to load.                                                                                                       |
+| `--config-directory <directory>` | Directory containing additional `*.conf` files                                                                                    |
+| `--watch-config`                 | Restart Telegraf on local config changes. Use either fs notifications (`inotify`) or polling (`poll`). Disabled by default        |
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:36]: does it make sense to spell out config for consistency, and use `config`only when referring to commands or file extension?

File: content/telegraf/v1.20/administration/commands.md (hunk 14 above)
Comment (kelseiv) [reply content/telegraf/v1.20/administration/commands.md:51]: would it be accurate to say service input plugins here? or add note to clarify that we refer to "configuration sections" as "plugins"...and "input plugins" as "inputs," or other terms we commonly use interchangeably?

File: content/telegraf/v1.20/administration/commands.md [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+telegraf config > telegraf.conf
+```
+
+### Generate config with only CPU input and InfluxDB output plugins defined
+
+```sh
+telegraf --input-filter cpu --output-filter influxdb config
+```
+
+### Run a single Telegraf collection, outputting metrics to stdout
```
Comment (kelseiv) [inline content/telegraf/v1.20/administration/commands.md:69]: is "collection" a commonly used term for all configured plugins?

**Review by kelseiv — APPROVED** [review]

File: content/telegraf/v1.20/administration/commands.md (hunk 19 above)
Comment (noramullen1) [reply content/telegraf/v1.20/administration/commands.md:69]: @sjwang90

Comment (sjwang90) [reply content/telegraf/v1.20/administration/commands.md:69]: Nah, I think collection should be "configuration"

**Pull request merged**